packages/catalog-client/src/types.ts

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Entity } from '@backstage/catalog-model';
+
+export type CatalogClient = {
+  entities(...options: EntitiesRequestOption[]): Promise<EntitiesResponse>;
+
+  entityByName(fullName: {
+    kind: string;
+    namespace: string;
+    name: string;
+  }): Promise<Entity | undefined>;
+};
+
+export type EntitiesRequestOption = {
+  apply(to: URLSearchParams): void;
```

Comment:
Yep, this 👌 

It may be wise to make this an object too, just forwards compatibility things. That way we could extend these to operate on other things than just the params in the future if we need to. `apply(to: { params: URLSearchParams })`, that is

Reply:
Still ofc not 100% sure this abstraction is worth it. It provides a lot more flexibility in evolving without breaking changes, but could end up in other issues, especially around testing.

Would be totally up for just trying to figure out a nice options object based approach too, even if we may end up wanting to break that API in the future.

Reply:
Yeah I at one point had a Request like wrapper instead of URLSearchParams (in case you want to add headers or cookies...) but it all started feeling so over complicated for something so essentially small :)